### ISSUE TITLE
With LDAP authenticator, unable to use local groups with Uppercase Characters

### ISSUE THREAD

**Issue by stvdilln** [hashicorp/vault#2623]
Ran across this in our version upgrade.  Local Groups with upper case characters were lost when people logged in.  

Vault writes to auth/domain/users/sd007 have the groups lower cased on the write.  When doing a login 
vault fails to find the appropriate groups because the group names are stored in whatever case 
is passed in.

```
steved@steve-mint ~ $ vault policy-write upper_policy test.hcl 
Policy 'upper_policy' written.
steved@steve-mint ~ $ vault policy-write lower_policy test.hcl 
Policy 'lower_policy' written.
steved@steve-mint ~ $ vault write auth/dynutil/groups/TEST_UPPER policies=upper_policy
Success! Data written to: auth/dynutil/groups/TEST_UPPER
steved@steve-mint ~ $ vault write auth/dynutil/groups/test_lower policies=lower_policy
Success! Data written to: auth/dynutil/groups/test_lower
steved@steve-mint ~ $  vault write auth/dynutil/users/sd0071 groups=TEST_UPPER,test_lower
Success! Data written to: auth/dynutil/users/sd0071
steved@steve-mint ~ $ vault auth -method=ldap username=sd0071 mount=dynutil
Password (will be hidden): 
Successfully authenticated! You are now logged in.
The token below is already saved in the session. You do not
need to "vault auth" again with the token.
token: 00f38a20-5a42-368f-6471-08b32030399c
token_duration: 2764799
token_policies: [default demo_human_policy  lower_policy]
steved@steve-mint ~ $ vault -v
Vault v0.7.0 ('614deacfca3f3b7162bbf30a36d6fc7362cd47f0')
steved@steve-mint ~ $ vault list auth/dynutil/groups | grep -i TEST_
TEST_UPPER
test_lower

vault read auth/dynutil/users/sd0071 
Key     	Value
---     	-----
groups  	test_lower,test_upper
policies	default

```

Expected Result:  At Login both lower_policy and upper_policy would be assigned to the user.  Upper policy is not found as there is a case mismatch between the list of local groups for the user and the values at auth/dyntuil/groups.
 
My suggestion would be that you do case insensitive searching for group names (and do not allow groups that only differ by case).

**Comment by chrishoffman** [comment]
Thanks for the report.  This is a duplicate of #2612 and addressed in #2613.  We are not going to do case-insensitive matching but will preserve case in the future when saving groups to users.

**Comment by stvdilln** [comment]
Thanks, sorry I didn't search before submitting a new issue.
